docker-compose.teamcity.yml

```diff
@@ -0,0 +1,20 @@
+version: '3.8'
```

Comment:
If you add the following env vars to the server & agent respectively - you'll enable debugging and also 'devmode'  which lets you inject JSP Files on the fly.
Will also need a port mapping for port 5010 on both containers.

TEAMCITY_SERVER_OPTS=-Dteamcity.development.mode=true -agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=*:5010

TEAMCITY_AGENT_OPTS=-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=5010

Reply:
it might be worth mounting the 'distributions' directory from the build area into the server container - the pluginsn are hotswappable, so you can just keep 'uploading' then 'enabling' without needing to restart the server/agent every time.

Reply:
Good tips, ill have a fiddle and see how far I can get. There was a bit of a clash between mounting the _whole_ data directory onto the host (to allow quicker restarting of the TC instance) and mounting the plugin directory _inside_ this directory. The approach you describe might be easier though, thanks I didn't know TeamCity would do that.

Reply:
was thinking you could mount distributions under /octopus/plugins - then in the TC UI, go to Admin --> Upload plugin.